hexrdgui/image_series_toolbar.py

```diff
@@ -74,14 +79,19 @@ def __init__(self, ims, parent=None):
 
     def setup_connections(self):
         self.slider.valueChanged.connect(self.val_changed)
-        self.slider.valueChanged.connect(self.frame.setValue)
-        self.frame.valueChanged.connect(
-            self.slider.setSliderPosition)
+        self.slider.valueChanged.connect(lambda i: self.frame.setText(str(i)))
+        self.frame.textChanged.connect(
+            lambda i: self.slider.setSliderPosition(int(i)))
```

Comment:
```suggestion
        self.frame.editingFinished.connect(self.on_frame_edited)
```

I think we should only trigger an update after the user has finished editing. Otherwise, if they type in `11`, for example, it triggers an update with every keystroke (so first loads frame `1` and then loads frame `11`).

But `editingFinished` also does not supply the new text as an argument, so you'd also have to check it with `self.frame.text()`.

Also, we need to do some validation for the text that the user enters. I can type in invalid text like `f` and it produces an exception. Instead, we should do something like this:

```python
    try:
        val = int(text)
    except ValueError:
        # Not a valid integer. Restore the previous value.
        val = prev_val

    # Clip the value to the min/max range
    val = max(min_val, val)
    val = min(max_val, val)
```

Notice also we should clip the value to the min/max range.